w3-store.md

```diff
@@ -1,631 +1,870 @@
+# W3 Storage Protocol
 
 ![status:wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
 
 ## Editors
 
 - [Yusef Napora](https://github.com/yusefnapora), [DAG House](https://dag.house)
+- [Irakli Gozalishvili](https://github.com/gozala)
 
 ## Authors
 
+- [Irakli Gozalishvili](https://github.com/gozala)
 - [Yusef Napora](https://github.com/yusefnapora), [DAG House](https://dag.house)
 
 ## Abstract
 
-This spec describes the data storage protocol used by web3.storage's "w3up" platform.
+In the W3 protocol user owned (name)space represent a data storage primitive that can be managed using W3 storage protocol defined here. Storage protocol allows space owners to manage state across compatible storage provider services using defined set of [UCAN] capabilities. Use of [UCAN] authorization system enables space access to be shared by delegating corresponding capabilities to desired audience.
 
-The protocol is modeled as a set of capabilities which can be fulfilled by a [provider](./w3-provider.md) via UCAN invocations. In the web3.storage implementation, the providers are implemented using [ucanto](https://github.com/web3-storage/ucanto), a type-safe UCAN RPC framework.
+## Language
 
-- [Background](#background)
-  - [Spaces](#spaces)
-- [Capabilities](#capabilities)
-  - [`store/` namespace](#store-namespace)
-    - [`store/*`](#store)
-    - [`store/add`](#storeadd)
-    - [`store/get`](#storeget)
-    - [`store/remove`](#storeremove)
-    - [`store/list`](#storelist)
-  - [`upload/` namespace](#upload-namespace)
-    - [`upload/*`](#upload)
-    - [`upload/add`](#uploadadd)
-    - [`upload/get`](#uploadget)
-    - [`upload/remove`](#uploadremove)
-    - [`upload/list`](#uploadlist)
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
-## Background
+# Introduction
 
-The w3up protocol accepts data in [CAR](https://ipld.io/specs/transport/car/) format, and clients are expected to encode data into CARs before invoking storage capabilities.
+The base storage layer of the space is [Content Address]ed [Content ARchive][CAR] files. In practice this means that user wishing to store files or directories of files needs produce an [IPLD] [DAG] in [UnixFS] format and then encoded into one or multiple [Content ARchive]s that can be stored using W3 storage protocol.
 
-The current w3up clients encode files and directories using [UnixFS](https://docs.ipfs.tech/concepts/file-systems/#unix-file-system-unixfs) and the [DAG-PB codec](https://ipld.io/specs/codecs/dag-pb/spec/), however, the storage protocol makes no assumptions about the format of the DAG (or DAGs) contained within a CAR, and any valid IPLD codec may be used.
+> Large DAGs get "sharded" across multiple [Content ARchive]s and stored individually to meet size limits of the storage provider.
```

Comment:
```suggestion
> Large DAGs get "sharded" across multiple [Content ARchive]s and stored individually to meet potential size limits of the storage provider.
```

well, by our experience with old.web3.storage, I think we also concluded that more shards can be better for paralelization and more efficient retries on error. In addition, this also enables building a pause and resume upload pipeline which is great use case. I would suggest expanding this as not only a potential limitation of the storage provider